src/App.tsx

```diff
@@ -1,26 +1,101 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodos, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TodoStatusFilter } from './types/TodoStatusFilter';
+import { getPreparedTodos } from './utils/getPreparedTodos';
+import { ErrorMessages } from './types/ErrorMessages';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState([0]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState<TodoStatusFilter>(
+    TodoStatusFilter.All,
+  );
+  const [loadingTodos, setLoadingTodos] = useState(true);
+
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedTodosCount = todos.filter(todo => todo.completed).length;
+  const preparedTodos = getPreparedTodos(todos, selectedFilter);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setIsLoading(prev => [...prev, todoId]);
+
+    deleteTodos(todoId)
+      .then(() =>
+        setTodos((prev: Todo[]) => prev.filter(todo => todo.id !== todoId)),
+      )
+      .catch(() => {
+        setErrorMessage(ErrorMessages.DeleteTodo);
```

Comment:
setErrorMessage(ErrorMessages.DeleteTodo);
Checklist 2. "Use destructuring wherever possible. It makes code more readable."